Application/Education/Certificate/Generator/Service/Data.php

```diff
@@ -31,941 +50,267 @@
-            $this->setCertificateSubject($tblCertificate, 'PH', 2, 4);
-            $this->setCertificateSubject($tblCertificate, 'SPO', 2, 5);
-            $this->setCertificateSubject($tblCertificate, 'REV', 2, 6);
-            $this->setCertificateSubject($tblCertificate, 'IN', 2, 7);
-        }
-        if ($tblCertificate) {
-            if ($tblSchoolTypeSecondary && $tblCourseMain) {
-                $this->updateCertificate($tblCertificate, $tblCertificateTypeDiploma, $tblSchoolTypeSecondary,
-                    $tblCourseMain);
-                if (!$this->getCertificateLevelAllByCertificate($tblCertificate)) {
-                    if (($tblLevel = Division::useService()->getLevelBy($tblSchoolTypeSecondary, '9'))) {
-                        $this->createCertificateLevel($tblCertificate, $tblLevel);
-                    }
-                }
-            }
-        }
+    /**
+     * @return TblConsumer
+     */
+    public function getTblConsumer()
+    {
+        return $this->tblConsumer;
+    }
 
-        $tblCertificate = $this->createCertificate('Mittelschule Abschlusszeugnis', 'Hauptschule qualifiziert',
-            'MsAbsHsQ');
-        if ($tblCertificate && !$this->getCertificateSubjectAll($tblCertificate)) {
-            $this->setCertificateSubject($tblCertificate, 'DE', 1, 1);
-            $this->setCertificateSubject($tblCertificate, 'EN', 1, 2);
-            $this->setCertificateSubject($tblCertificate, 'KU', 1, 3);
-            $this->setCertificateSubject($tblCertificate, 'MU', 1, 4);
-            $this->setCertificateSubject($tblCertificate, 'GE', 1, 5);
-            $this->setCertificateSubject($tblCertificate, 'GK', 1, 6);
-            $this->setCertificateSubject($tblCertificate, 'GEO', 1, 7);
-            $this->setCertificateSubject($tblCertificate, 'WTH', 1, 8);
-
-            $this->setCertificateSubject($tblCertificate, 'MA', 2, 1);
-            $this->setCertificateSubject($tblCertificate, 'BIO', 2, 2);
-            $this->setCertificateSubject($tblCertificate, 'CH', 2, 3);
-            $this->setCertificateSubject($tblCertificate, 'PH', 2, 4);
-            $this->setCertificateSubject($tblCertificate, 'SPO', 2, 5);
-            $this->setCertificateSubject($tblCertificate, 'REV', 2, 6);
-            $this->setCertificateSubject($tblCertificate, 'IN', 2, 7);
-        }
-        if ($tblCertificate) {
-            if ($tblSchoolTypeSecondary && $tblCourseMain) {
-                $this->updateCertificate($tblCertificate, $tblCertificateTypeDiploma, $tblSchoolTypeSecondary,
-                    $tblCourseMain);
-                if (!$this->getCertificateLevelAllByCertificate($tblCertificate)) {
-                    if (($tblLevel = Division::useService()->getLevelBy($tblSchoolTypeSecondary, '9'))) {
-                        $this->createCertificateLevel($tblCertificate, $tblLevel);
-                    }
-                }
-            }
-        }
+    public function setupDatabaseContent()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method setupDatabaseContent() has a Cyclomatic Complexity of 523. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5646354)